src/main/java/com/derongan/minecraft/deeperworld/world/Point.java

```diff
@@ -35,4 +41,24 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(getX(), getZ());
     }
+
+    public Point plus(Point other){
```

Comment:
This looks like the perfect place to use kotlin's operator functions and a data class to auto generate hashCode and whatnot.

Reply:
Oh sorry, didn't realize that kotlin lets you do that from Java. This looks good as is then!